features/post-meta.feature

```diff
@@ -139,3 +139,273 @@ Feature: Manage post custom fields
+    And STDERR should contain:
+      """
+      No data exists for not-a-key
+      """
+    And the return code should be 1
+
+  @patch @patch-delete
+  Scenario: A key can be deleted from a nested value.
+    Given a WP install
+    And an input.json file:
+      """
+      {
+        "foo": {
+          "bar": "baz",
+          "abe": "lincoln"
+        }
+      }
+      """
+    And I run `wp post meta set 1 meta-key --format=json < input.json`
+
+    When I run `wp post meta patch delete 1 meta-key foo bar`
+    Then STDOUT should be:
+      """
+      Success: Updated custom field 'meta-key'.
+      """
+
+    When I run `wp post meta get 1 meta-key --format=json`
+    Then STDOUT should be JSON containing:
+      """
+      {
+        "foo": {
+          "abe": "lincoln"
+        }
+      }
+      """
+
+  @patch @patch-fail @patch-delete @patch-delete-fail
+  Scenario: A key cannot be deleted from a nested value from a non-existent key.
+    Given a WP install
+    And an input.json file:
+      """
+      {
+        "foo": {
+          "bar": "baz"
+        }
+      }
+      """
+    And I run `wp post meta set 1 meta-key --format=json < input.json`
+
+    When I try `wp post meta patch delete 1 meta-key foo not-a-key`
+    Then STDOUT should be empty
+    And STDERR should contain:
+      """
+      No data exists for not-a-key
+      """
+    And the return code should be 1
+
+  @patch @patch-insert
+  Scenario: A new key can be inserted into a nested value.
+    Given a WP install
+    And I run `wp post meta set 1 meta-key '{}' --format=json`
+
+    When I run `wp post meta patch insert 1 meta-key foo bar`
+    Then STDOUT should be:
+      """
+      Success: Updated custom field 'meta-key'.
+      """
+
+    When I run `wp post meta get 1 meta-key --format=json`
+    Then STDOUT should be JSON containing:
+      """
+      {
+        "foo": "bar"
+      }
+      """
+
+  @patch @patch-fail @patch-insert @patch-insert-fail
+  Scenario: A new key cannot be inserted into a non-nested value.
+    Given a WP install
+    And I run `wp post meta set 1 meta-key 'a simple value'`
+
+    When I try `wp post meta patch insert 1 meta-key foo bar`
+    Then STDOUT should be empty
+    And STDERR should contain:
+      """
+      Cannot create key 'foo'
+      """
+    And the return code should be 1
+
+    When I run `wp post meta get 1 meta-key`
+    Then STDOUT should be:
+      """
+      a simple value
+      """
+
+  @patch @patch-numeric
+  Scenario: A nested value can be updated using an integer key.
+    Given a WP install
+    And I run `wp post meta set 1 meta-key '[ "foo", "bar" ]' --format=json`
+
+    When I run `wp post meta patch update 1 meta-key 0 new`
+    Then STDOUT should be:
+      """
+      Success: Updated custom field 'meta-key'.
+      """
+
+    When I run `wp post meta get 1 meta-key --format=json`
+    Then STDOUT should be JSON containing:
+      """
+      [ "new", "bar" ]
+      """
```

Comment:
> What happens when you patch insert an already existing value?

Will add.

> What happens when you patch update a non-existing key? (This should produce an error.)

I believe this is covered already. See the `@patch-fail` tags.

> What happens when you patch update an existing key to the same value? (You seem to have an adapted success message in the code below.)

Will add. (Should be a notice that value did not change.)